lib/pure/parsejson.nim

```diff
@@ -524,3 +599,31 @@ proc raiseParseErr*(p: JsonParser, msg: string) {.noinline, noreturn.} =
 proc eat*(p: var JsonParser, tok: TokKind) =
   if p.tok == tok: discard getTok(p)
   else: raiseParseErr(p, tokToStr[tok])
+
+iterator jsonTokens*(input: Stream, path: string, rawStringLiterals = false,
+                     strIntegers = true, strFloats = true): ptr JsonParser =
+  ## Yield each successive `ptr JsonParser` while parsing `input` with error
+  ## label `path`.
+  var p: JsonParser
+  p.open(input, path, rawStringLiterals = rawStringLiterals,
+         strIntegers = strIntegers, strFloats = strFloats)
+  try:
+    var tok = p.getTok
+    while tok notin {tkEof,tkError}:
+      yield p.addr                      # `JsonParser` is a pretty big struct
+      tok = p.getTok
+  finally:
+    p.close()
+
+iterator jsonTokens*(path: string, rawStringLiterals = false,
```

Comment:
These API additions are not good for two reasons:

- An iterator that always produces the same value is a foreign concept.
- It looks like they are unused. :-)

Reply:
They're unused because the idea was to provide something like `json.parseJsonFragments` but in the raw parsing `parsejson` context.  I think they are *also* not normal in yielding a `ptr JsonParser`, but that actually made like 1.5X performance difference with the new faster logic because the JsonParser object is big enough the alloc/copies showed up a the top of the profile.  It mostly seemed the easiest way to really streamline the usage syntax, as in [here](https://forum.nim-lang.org/t/6920#44912).  I'm open to alternative suggestions to streamline said syntax, or just removing them and leaving them to the user, though.

I was more concerned about your feedback about `int64` but not `uint64` overflow ideas, going down the `options = set[enum]` route and adding a new `u: uint64` field for users to conditionally/unconditionally access when they maybe/definitely expect unsigned, and that sort of direction.  That's much more to do/get right than, say, moving those iterators to example code/doing something else.  So, if you hated the general direction I wanted to hold off on that work.

Reply:
The return type could be ``lent T``  but I don't know of today's compiler accepts that (it really shouldn't without a spec addition...).

For now please remove these iterators, it's easy enough to write a open/close/loop snippet.

> I was more concerned about your feedback about int64 but not uint64 overflow ideas, going down the options = set[enum] route and adding a new u: uint64 field for users to conditionally/unconditionally access when they maybe/definitely expect unsigned, and that sort of direction. 

I would use a single ``int64`` field and then accessor procs/templates that hide the ``cast[uint64](x)``. And ``options = set[enum]`` is the better design yes. I couldn't use it because of backwards compatibility reasons. There is a different solution that deals with the compat problems but it's annoying to do, maybe it's time to make ``.since`` a builtin that works better.